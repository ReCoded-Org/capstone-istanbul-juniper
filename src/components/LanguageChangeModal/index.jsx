import React from "react";
import { Modal } from "antd";
import i18n from "i18next";

const LanguageChangeModal = ({
  modalVisibility,
  setModalVisibility,
  newLanguage,
}) => {
  return (
    <Modal
      title="Basic Modal"
      visible={modalVisibility}
      onOk={() => {
        setModalVisibility(false);
        i18n.changeLanguage(newLanguage);
        window.location.reload();
      }}
      onCancel={() => {
        setModalVisibility(false);
      }}
    >
      <p>Changing language will cause page to reload.</p>
      <p>Are you sure?</p>
    </Modal>
  );
};

export default LanguageChangeModal;
