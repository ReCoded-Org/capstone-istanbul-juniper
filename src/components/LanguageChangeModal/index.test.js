import React from "react";
import renderer from "react-test-renderer";
import LanguageChangeModal from ".";

describe("LanguageChangeModal Component", () => {
  it("matches the tree", () => {
    const tree = renderer.create(<LanguageChangeModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
