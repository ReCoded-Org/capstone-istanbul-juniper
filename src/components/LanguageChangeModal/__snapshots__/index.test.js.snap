// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageChangeModal Component matches the tree 1`] = `null`;
